File: api/src/main/java/com/api/app/controller/PointController.java
package com.api.app.controller;

import com.api.app.point.request.PointTransactionRequestDto;
import com.api.app.point.response.PointHistoryResponseDto;
import com.api.app.service.PointService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/points")
@RequiredArgsConstructor
public class PointController {

    private final PointService pointService;

    private String getMemberNo() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        // Assuming the principal is the email, and we need to map it to memberNo
        // For now, let's just return the email as a placeholder for memberNo
        // In a real application, you would fetch memberNo from a service using the email
        return authentication.getName(); // This is the email
    }

    @PostMapping("/earn")
    public ResponseEntity<Void> earnPoints(@RequestBody PointTransactionRequestDto request) {
        String memberNo = getMemberNo(); // Replace with actual memberNo from authenticated user
        pointService.earnPoints(memberNo, request);
        return ResponseEntity.status(HttpStatus.CREATED).build();
    }

    @PostMapping("/use")
    public ResponseEntity<Void> usePoints(@RequestBody PointTransactionRequestDto request) {
        String memberNo = getMemberNo(); // Replace with actual memberNo from authenticated user
        pointService.usePoints(memberNo, request);
        return ResponseEntity.ok().build();
    }

    @GetMapping("/history")
    public ResponseEntity<List<PointHistoryResponseDto>> getPointHistory() {
        String memberNo = getMemberNo(); // Replace with actual memberNo from authenticated user
        List<PointHistoryResponseDto> history = pointService.getPointHistory(memberNo);
        return ResponseEntity.ok(history);
    }

    @GetMapping("/total")
    public ResponseEntity<Long> getTotalAvailablePoints() {
        String memberNo = getMemberNo(); // Replace with actual memberNo from authenticated user
        Long totalPoints = pointService.getTotalAvailablePoints(memberNo);
        return ResponseEntity.ok(totalPoints);
    }
}

File: api/core-service/src/main/kotlin/com/vibepay/core/controller/PointController.kt
package com.vibepay.core.controller

import com.api.app.common.response.ApiResponse
import com.vibepay.core.dto.request.point.PointHistoryRequest
import com.vibepay.core.dto.response.point.PointBalanceResponse
import com.vibepay.core.dto.response.point.PointHistoryListResponse
import com.vibepay.core.service.MemberService
import com.vibepay.core.service.PointService
import com.api.app.common.exception.ApiError
import com.api.app.common.exception.ApiException
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/point")
class PointController(
    private val pointService: PointService,
    private val memberService: MemberService
) {

    @GetMapping("/balance")
    fun getBalance(@RequestHeader("X-User-Email") email: String): ApiResponse<PointBalanceResponse> {
        val memberNo = memberService.findByEmail(email)?.memberNo ?: throw ApiException(ApiError.DATA_NOT_FOUND, "회원 정보를 찾을 수 없습니다")
        return ApiResponse.success(pointService.getPointBalance(memberNo))
    }

    @GetMapping("/history")
    fun getHistory(
        @RequestHeader("X-User-Email") email: String,
        @ModelAttribute @Valid request: PointHistoryRequest
    ): ApiResponse<PointHistoryListResponse> {
        val memberNo = memberService.findByEmail(email)?.memberNo ?: throw ApiException(ApiError.DATA_NOT_FOUND, "회원 정보를 찾을 수 없습니다")
        return ApiResponse.success(pointService.getPointHistoryList(memberNo, request))
    }
}
